application/common/models/User.php

```diff
@@ -343,6 +343,55 @@ private function savePassword()
         return true;
     }
 
+    public static function getExpiringUsers($criteria): array
+    {
+        if (empty($criteria)) {
+            return [];
+        }
+
+        $users = User::find()->joinWith('currentPassword')
+                             ->where(['active' => 'yes']);
+
+        foreach ($criteria as $name => $value) {
+            switch ($name) {
+                case 'expires_on':
+                case 'grace_period_ends_on':
+                    $users->andWhere(["password.$name" => $value]);
+                    break;
+                default:
+                    // if no criteria names match, this will ensure an empty result is returned
+                    $users->where('0=1');
+            }
+        }
+
+        return $users->all();
+    }
+
+    public static function getNewUsers($createdOn): array
```

Comment:
That is the intent for now until we find a need to include the idea of newly created users in our definition of "new"